sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -253,6 +253,36 @@ def fillna(self, value, method, axis, limit, **kwargs):
   backfill = _fillna_alias('backfill')
   pad = _fillna_alias('pad')
 
+  @frame_base.with_docs_from(pd.DataFrame)
+  def first(self, offset):
+    per_partition = expressions.ComputedExpression(
+        'first-per-partition',
+        lambda df: df.sort_index().first(offset=offset), [self._expr],
+        preserves_partition_by=partitionings.Arbitrary(),
```

Comment:
Assuming this means that it preserves no partitioning?

Reply:
This actually means it will preserve any partitioning, `preserves=Singleton()` would indicate it preserves no partitioning.

In this case the operation doesn't modify the index at all, so the output should still be partitioned in the same way.